ocaml/typing/typecore.ml

```diff
@@ -3767,6 +3774,16 @@ and type_approx_aux_jane_syntax : Jane_syntax.Expression.t -> _ = function
   | Jexp_immutable_array _
   | Jexp_unboxed_constant _ -> ()
 
+and type_approx_aux_case_rhs env rhs in_function ty_expected =
+  match rhs with
+  | Psimple_rhs e | Pboolean_guarded_rhs { pbg_rhs = e; _ } ->
+      type_approx_aux env e in_function ty_expected
+  | Ppattern_guarded_rhs { ppg_cases; _ } ->
+      match ppg_cases with
+      | [] -> ()
+      | { pc_rhs; _ } :: _ ->
+          type_approx_aux_case_rhs env pc_rhs in_function ty_expected
```

Comment:
I know this is a nit, but I wonder if we can share even *more* code with the `Pexp_match` case by writing a `type_approx_aux_cases` function &mdash; it would be a shame to forget to update one place or the other because we forgot they were related, and a helper function obviates this drift.

Reply:
Nits are good, they make the code better :)

Made this change!